python/cudf/cudf/options.py

```diff
@@ -58,6 +67,10 @@ def _register_option(
     validator : Callable
         Called on the option value to check its validity. Should raise an
         error if the value is invalid.
+    set_callback : Callable | None
+        Called when setting the option value.
+    get_callback : Callable | None
+        Called when getting the option value.
```

Comment:
Do we really need `set_callback` and `get_callback` here?

We have other places where we control behavior, especially global state, outside of cudf Python options. Examples include the RMM memory resource and numba options. It is awkward for a cuDF option to alter the global state of another library (KvikIO). I would advocate for users to call directly through pylibcudf if this setting is specific to libcudf, or call through KvikIO Python APIs if they want to alter some global state in kvikio.

Reply:
Wouldn't it be better if KvikIO's Python API could be used to manage KvikIO options?

cuDF's options system shouldn't be the way that users modify the behavior of pylibcudf or cudf-polars.

Reply:
The challenge with directly using kvikIO APIs to set the global state is that libcudf is doing the same, so the final state depends on the order of operations.

But I think I do see the incongruity in the API design. Perhaps rebranding this as "IO settings" would make it a better fit in libcudf. kvikIO becomes an implementation detail, and the library that uses libcudf does not need to know how the IO is performed.